lnrpc/invoicesrpc/invoices.proto

```diff
@@ -192,3 +201,45 @@ message LookupInvoiceMsg {
 
     LookupModifier lookup_modifier = 4;
 }
+
+// CircuitKey is a unique identifier for an HTLC.
+message CircuitKey {
+    /// The id of the channel that the is part of this circuit.
+    uint64 chan_id = 1;
+
+    /// The index of the incoming htlc in the incoming channel.
+    uint64 htlc_id = 2;
+}
+
+message HtlcModifyRequest {
+    // The invoice the intercepted HTLC is attempting to settle. The HTLCs in
+    // the invoice are only HTLCs that have already been accepted or settled,
+    // not including the current intercepted HTLC.
+    lnrpc.Invoice invoice = 1;
+
+    // The unique identifier of the HTLC of this intercepted HTLC.
+    CircuitKey exit_htlc_circuit_key = 2;
+
+    // The amount in milli-satoshi that the exit HTLC is attempting to pay.
+    uint64 exit_htlc_amt = 3;
+
+    // The absolute expiry height of the exit HTLC.
+    uint32 exit_htlc_expiry = 4;
+
+    // The current block height.
+    uint32 current_height = 5;
+
+    // The wire message custom records of the exit HTLC.
+    map<uint64, bytes> exit_htlc_wire_custom_records = 6;
+}
+
+message HtlcModifyResponse {
```

Comment:
my only side note in this PR is that we may wanna add more stuff here in the future?

Reply:
Yeah, I guess we might want to, but it's not clear right now if we actually need to. So leaving off a TODO for now to avoid needing to trigger another CI run. But added this comment to the tracking issue.